### PR TITLE
Migrate from Foreman::Model::GCE to ForemanGoogle::GCE

### PR DESCRIPTION
**Pull request by stejskalleos**

### CODE REVIEW

File: db/migrate/20220331113745_foreman_gce_to_foreman_google_gce.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class ForemanGceToForemanGoogleGce < ActiveRecord::Migration[6.0]
+  def up
+    User.without_auditing do
+      old_type = 'Foreman::Model::GCE'
+      new_type = 'ForemanGoogle::GCE'
+      compute_resources = ComputeResource.unscoped.where(type: old_type)
+
+      compute_resources.each do |cr|
+        json_data = JSON.parse(File.read(cr.attrs[:key_path]))
```
Comment (ezr-ondrej) [inline db/migrate/20220331113745_foreman_gce_to_foreman_google_gce.rb]: You do not need to parse the `JSON`, we are storing the string.

File: db/migrate/20220331113745_foreman_gce_to_foreman_google_gce.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  def down
+    User.without_auditing do
+      old_type = 'ForemanGoogle::GCE'
+      new_type = 'Foreman::Model::GCE'
+      compute_resources = ComputeResource.unscoped.where(type: old_type)
+
+      compute_resources.each do |cr|
+        cr.update(type: new_type, password: nil)
+      end
```
Comment (ezr-ondrej) [inline db/migrate/20220331113745_foreman_gce_to_foreman_google_gce.rb]:
```suggestion
      compute_resources.update_all(type: new_type, password: nil)
```

**Review by ezr-ondrej — APPROVED** [review]
Looks ok to me, all below are just nits :+1:

For up migration we need to deserialize attrs to read the key_path, but for the down, we don't have that need, can we just update all at unce and not load the CRs? :)

This is diffucult, so I would just go with whatever we feel like, but is it actually worth down migrating? we might end up with broken GCE compute resource as down migration can mean someone is uninstalling the plugin, but we might have already removed GCE from core :shrug:

**Comment by stejskalleos** [comment]
> This is diffucult, so I would just go with whatever we feel like, but is it actually worth down migrating? we might end up with broken GCE compute resource as down migration can mean someone is uninstalling the plugin, but we might have already removed GCE from core

Yeah make sense, we can make more problems than solutions

**Pull request merged**